src/goSuggest.ts

```diff
@@ -44,6 +44,14 @@ export class GoCompletionItemProvider implements vscode.CompletionItemProvider {
 		return this.provideCompletionItemsInternal(document, position, token, vscode.workspace.getConfiguration('go'));
 	}
 
+	private isInBlockComment(document: vscode.TextDocument, position: vscode.Position): boolean {
+		let text = document.getText();
+		let currentOffset = document.offsetAt(position);
+		let lastBlockCommentStartIndex = text.lastIndexOf('/*', currentOffset);
+		let lastBlockCommentEndIndex = text.lastIndexOf('*/', currentOffset);
+		return lastBlockCommentStartIndex > lastBlockCommentEndIndex;
```

Comment:
It would definitely be nice to have the AST in hand during completions

Apart from stopping  completions inside multi line comments and multi line strings, it would also help in #168 

> But I don't know how well the parser can deal with incomplete source code...

I dont know much here either. 

We currently use https://github.com/ramya-rao-a/go-outline to get the file outline which uses the go parser. You can try and see if we can re-use that here